File: app/src/main/java/com/freetime/geoweather/ui/LocationsViewModel.java
package com.freetime.geoweather.ui;

import android.app.Application;

import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import com.freetime.geoweather.data.LocationDao;
import com.freetime.geoweather.data.LocationDatabase;
import com.freetime.geoweather.data.LocationEntity;

import java.util.List;

public class LocationsViewModel extends AndroidViewModel {

    private final LocationDao locationDao;
    public final LiveData<List<LocationEntity>> locations;

    public LocationsViewModel(Application application) {
        super(application);
        LocationDatabase db = LocationDatabase.getDatabase(application);
        locationDao = db.locationDao();
        locations = locationDao.getAllLocations();
    }

    public void addLocation(String name, double latitude, double longitude) {
        LocationDatabase.databaseWriteExecutor.execute(() -> {
            LocationEntity newLocation = new LocationEntity(name, latitude, longitude);
            locationDao.insertLocation(newLocation);
        });
    }

    public void deleteLocation(LocationEntity location) {
        LocationDatabase.databaseWriteExecutor.execute(() -> {
            locationDao.deleteLocation(location);
        });
    }
}

File: app/src/main/java/com/freetime/geoweather/ui/LocationsViewModel.kt
package com.freetime.geoweather.ui

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import com.freetime.geoweather.data.LocationDao
import com.freetime.geoweather.data.LocationDatabase
import com.freetime.geoweather.data.LocationEntity

class LocationsViewModel(application: Application) : AndroidViewModel(application) {

    private val locationDao: LocationDao = LocationDatabase.getDatabase(application).locationDao()
    val locations: LiveData<List<LocationEntity>> = locationDao.getAllLocations()

    fun addLocation(name: String, latitude: Double, longitude: Double) {
        LocationDatabase.databaseWriteExecutor.execute {
            val newLocation = LocationEntity(name = name, latitude = latitude, longitude = longitude)
            locationDao.insertLocation(newLocation)
        }
    }

    fun deleteLocation(location: LocationEntity) {
        LocationDatabase.databaseWriteExecutor.execute {
            locationDao.deleteLocation(location)
        }
    }
}